genai-perf/genai_perf/measurements/run_config_measurement.py

```diff
@@ -0,0 +1,660 @@
+        """
+        Sets the model weightings used when calculating
+        weighted metrics
+
+        Weights are the relative importance of the model
+        with respect to one another
+        """
+        assert model_weights.keys() == self._model_config_measurements.keys()
+
+        self._model_weights = {
+            model_name: model_weight / sum(model_weights.values())
+            for model_name, model_weight in model_weights.items()
+        }
+
+    def set_gpu_metric_objectives(
+        self, gpu_metric_objectives: Dict[ModelName, MetricObjectives]
+    ) -> None:
+        """
+        Sets the metric weighting for all GPU metric based measurements
+        """
+        for model_name in gpu_metric_objectives.keys():
+            assert model_name in self._model_weights.keys()
+
+        for model_name, model_gpu_metric_objectives in gpu_metric_objectives.items():
+            self._gpu_metric_objectives[model_name] = {
+                objective: (value / sum(model_gpu_metric_objectives.values()))
+                for objective, value in model_gpu_metric_objectives.items()
+            }
+
+    def set_perf_metric_objectives(
+        self, perf_metric_objectives: Dict[ModelName, MetricObjectives]
+    ) -> None:
+        """
+        Sets the metric weighting for all perf metric based measurements
+        """
+        for key in perf_metric_objectives.keys():
+            assert key in self._model_config_measurements.keys()
+
+        for model_name, metric_objectives in perf_metric_objectives.items():
+            self._model_config_measurements[model_name].set_metric_objectives(
+                metric_objectives
+            )
+
+    def set_constraints(self, constraints: RunConstraints) -> None:
+        self._constraints = constraints
+
+    def add_model_config_measurement(
+        self,
+        model_name: ModelName,
+        perf_metrics: PerfRecords,
+    ) -> None:
+        """
+        Adds the perf metrics obtained from profiling a single model
+        """
+        self._model_config_measurements[model_name] = ModelConfigMeasurement(
+            perf_metrics
+        )
+        self._model_weights[model_name] = RunConfigMeasurementDefaults.MODEL_WEIGHTING
+
+    ###########################################################################
+    # Checkpoint Methods
+    ###########################################################################
+    def write_to_checkpoint(self) -> Dict[str, Any]:
+        """
+        Converts the class data into a dictionary that can be written to
+        the checkpoint file
+        """
+        rcm_dict = deepcopy(self.__dict__)
+
+        # Values based solely on user/config settings (that can vary from run to run)
+        # are not stored in the checkpoint
+        del rcm_dict["_model_weights"]
+        del rcm_dict["_constraints"]
+
+        return rcm_dict
+
+    @classmethod
+    def read_from_checkpoint(cls, rcm_dict: Dict[str, Any]) -> "RunConfigMeasurement":
```

Comment:
This is to pass type checking. The return type is actually an instance of the RCM class, but since this class is declared in the same file, this is how you get around that issue.

Reply:
got it. thanks